Marlin/src/HAL/HAL_LPC1768/LPC1768_PWM.cpp

```diff
@@ -24,73 +24,78 @@
  * The class Servo uses the PWM class to implement its functions
  *
  * All PWMs use the same repetition rate - 20mS because that's the normal servo rate
-*/
+ */
 
 /**
  * This is a hybrid system.
  *
- * The PWM1 module is used to directly control the Servo 0, 1 & 3 pins.  This keeps
+ * The PWM1 module is used to directly control the Servo 0, 1 & 3 pins and D9 & D10 pins.  This keeps
  * the pulse width jitter to under a microsecond.
  *
  * For all other pins the PWM1 module is used to generate interrupts.  The ISR
  * routine does the actual setting/clearing of pins.  The upside is that any pin can
  * have a PWM channel assigned to it.  The downside is that there is more pulse width
  * jitter. The jitter depends on what else is happening in the system and what ISRs
- * prempt the PWM ISR.  Writing to the SD card can add 20 microseconds to the pulse
- * width.
+ * pre-empt the PWM ISR.
  */
 
 /**
  * The data structures are setup to minimize the computation done by the ISR which
- * minimizes ISR execution time.  Execution times are 2.2 - 3.7 microseconds.
+ * minimizes ISR execution time.  Execution times are 2-4 microseconds except when
+ * updating to a new value when they are 19 uS.
+ *
+ * Two tables are used.  One table contains the data used by the ISR to update/control
+ * the PWM pins.  The other is used as an aid when rebuilding the ISR table.
  *
- * Two tables are used.  active_table is used by the ISR.  Changes to the table are
- * are done by copying the active_table into the work_table, updating the work_table
- * and then swapping the two tables.  Swapping is done by manipulating pointers.
+ * The LPC1768_PWM_attach_pin routine disables the ISR and then adds the new info to
+ * ISR table.  It can update the table directly because none of its changes affect
+ * what the ISR does.
  *
- * Immediately after the swap the ISR uses the work_table until the start of the
- * next 20mS cycle. During this transition the "work_table" is actually the table
- * that was being used before the swap.  The "active_table" contains the data that
- * will start being used at the start of the next 20mS period.  This keeps the pins
- * well behaved during the transition.
+ * LPC1768_PWM_detach_pin routine disables the ISR, disables the pin immediately if
+ * it's a directly controlled pin and updates the helper table.  It then flags the
+ * ISR that the ISR table needs to be rebuilt.
  *
- * The ISR's priority is set to the maximum otherwise other ISRs can cause considerable
- * jitter in the PWM high time.
+ * LPC1768_PWM_write routine disables the ISR and updates the helper table.  It then
+ * flags the ISR that the ISR table needs to be rebuilt.
+ *
+ * The ISR's priority is set to less than the stepper ISR otherwise it could cause jitter
+ * in the step pulses.
  *
  * See the end of this file for details on the hardware/firmware interaction
  */
 
-
 #ifdef TARGET_LPC1768
+
+#include <src/Marlin.h>
```

Comment:
Any way to avoid this dependency? And, for project includes I think it's important to use the form
```cpp
#include "../../src/Marlin.h"
```
…and reserve the use of pointy braces for Sketchbook/library includes.